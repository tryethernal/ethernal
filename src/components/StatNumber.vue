<template>
    <v-card outlined>
        <v-card-subtitle>
            <div :class="{ absolute: infoTooltip }">{{ title }}</div>
            <div class="text-right" v-if="infoTooltip">
                <v-tooltip left>
                    <template v-slot:activator="{ on, attrs }">
                        <v-icon v-bind="attrs" v-on="on" small>mdi-information</v-icon>
                    </template>
                    {{ infoTooltip }}
                </v-tooltip>
            </div>
        </v-card-subtitle>
        <v-card-text class="text-h3" align="center">
            <v-skeleton-loader v-if="loading" type="list-item"></v-skeleton-loader>
            <template v-else-if="isValueDefined">
                <router-link v-if="type == 'link'" style="text-decoration: none;" :to="href">{{ commify(value) }}</router-link>
                <span v-else>
                    {{ formatNumber(value, { short: !long, decimals: decimals }) }}
                </span>
            </template>
            <template v-else>
                N/A
            </template>
        </v-card-text>
    </v-card>
</template>

<script>
const ethers = require('ethers');
const { formatNumber } = require('../lib/utils');

export default {
    name: 'StatNumber',
    props: ['type', 'title', 'value', 'loading', 'href', 'infoTooltip', 'decimals', 'long'],
    methods: {
        commify: ethers.utils.commify,
        formatNumber: formatNumber
    },
    computed: {
        isValueDefined() {
            return this.value !== undefined && this.value !== null;
        }
    }
}
</script>

<style scoped>
.absolute {
    position: absolute;
}
</style>
