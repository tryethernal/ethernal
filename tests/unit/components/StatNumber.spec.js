import flushPromises from 'flush-promises';
import MockHelper from '../MockHelper';

import StatNumber from '@/components/StatNumber.vue';

describe('StatNumber.vue', () => {
    let helper;

    beforeEach(async () => {
        helper = new MockHelper();
    });

    it('Should show a link', () => {
        const wrapper = helper.mountFn(StatNumber, {
            propsData: {
                type: 'link',
                title: 'Stat',
                value: 1,
                loading: false,
                href: '/blocks/1'
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should show the full number', () => {
        const wrapper = helper.mountFn(StatNumber, {
            propsData: {
                title: 'Stat',
                value: "83742318000000000000000000",
                loading: false,
                long: true
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should show a number', () => {
        const wrapper = helper.mountFn(StatNumber, {
            propsData: {
                title: 'Stat',
                value: 1,
                loading: false
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should show the info tooltip', () => {
        const wrapper = helper.mountFn(StatNumber, {
            propsData: {
                title: 'Stat',
                value: 1,
                loading: false,
                infoTooltip: 'Info Tooltip'
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should show the loading skeleton', async () => {
        const wrapper = helper.mountFn(StatNumber, {
            propsData: {
                title: 'Stat',
                value: 1,
                loading: true,
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
