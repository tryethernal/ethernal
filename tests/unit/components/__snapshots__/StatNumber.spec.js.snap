// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatNumber.vue Should show a link 1`] = `
<div class="v-card v-sheet v-sheet--outlined theme--light">
  <div class="v-card__subtitle">
    <div class="">Stat</div>
    <!---->
  </div>
  <div align="center" class="v-card__text text-h3"><a href="#/blocks/1" class="" style="text-decoration: none;">1</a></div>
</div>
`;

exports[`StatNumber.vue Should show a number 1`] = `
<div class="v-card v-sheet v-sheet--outlined theme--light">
  <div class="v-card__subtitle">
    <div class="">Stat</div>
    <!---->
  </div>
  <div align="center" class="v-card__text text-h3"><span>
                1
            </span></div>
</div>
`;

exports[`StatNumber.vue Should show the full number 1`] = `
<div class="v-card v-sheet v-sheet--outlined theme--light">
  <div class="v-card__subtitle">
    <div class="">Stat</div>
    <!---->
  </div>
  <div align="center" class="v-card__text text-h3"><span>
                83,742,318
            </span></div>
</div>
`;

exports[`StatNumber.vue Should show the info tooltip 1`] = `
<div class="v-card v-sheet v-sheet--outlined theme--light">
  <div class="v-card__subtitle">
    <div class="absolute">Stat</div>
    <div class="text-right"><span class="v-tooltip v-tooltip--left"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-information theme--light" style="font-size: 16px;"></i></span></div>
  </div>
  <div align="center" class="v-card__text text-h3"><span>
                1
            </span></div>
</div>
`;

exports[`StatNumber.vue Should show the loading skeleton 1`] = `
<div class="v-card v-sheet v-sheet--outlined theme--light">
  <div class="v-card__subtitle">
    <div class="">Stat</div>
    <!---->
  </div>
  <div align="center" class="v-card__text text-h3">
    <div aria-busy="true" aria-live="polite" role="alert" class="v-skeleton-loader v-skeleton-loader--is-loading theme--light">
      <div class="v-skeleton-loader__list-item v-skeleton-loader__bone">
        <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
      </div>
    </div>
  </div>
</div>
`;
